src/Database/LSMTree/Internal.hs

```diff
@@ -1124,6 +1124,8 @@ data TableConfig = TableConfig {
     -- | Function for resolving 'Mupsert' values. This should be 'Nothing' for
     -- normal tables and 'Just' for monoidal tables.
   , confResolveMupsert   :: !(Maybe ResolveMupsert)
+    -- | Bypass the OS page cache as much as possible during disk lookups.
+  , confNoCacheLookups   :: !Bool
```

Comment:
TODO: generalise this later to have a threshold so that (runs in) levels below a certain size will use the page cache, and (runs in) levels above that will not use the page cache. Or a three-way option: all, threshold, none. This allows applications that know they have some temporal locality to take advantage of that.

Apparently Cardano has somewhat more temporal locality than bitcoin and a reasonably high portion of inserts are looked up and deleted fairly soon after their creation. This would correspond to a reasonable cache hit rate in the smallest level(s).

Reply:
```
data DataCachingPolicy =

       -- | Use the OS page cache to cache any\/all data in all levels.
       --
       -- Use this policy if the expected access pattern for the table
       -- has a good spatial or temporal locality.
       DataCachingAllLevels
       
       -- | Use the OS page cache to cache data in all levels below a given
       -- size threshold.
       --
       -- Use this policy if the expected access pattern for the table
       -- has good temporal locality for recently inserted keys.
     | DataCachingLevelsBelow Int
       --TODO: decide and document the exact interpretation of the size.
       -- Is it in bytes? Is it cumulative down the levels? An easy use
       -- configuration option would be to say: "cache the first 10 Mb" and
       -- have everything worked out from that.
     
       -- | Do not cache any data in any level (except the write buffer).
       --
       -- Use this policy if expected access pattern for the table has poor
       -- spatial or temporal locality, such as uniform random access.
     | DataCachingNone
```